java/feathub-udf/flink-udf/src/main/java/com/alibaba/feathub/flink/udf/JavaUdfUtils.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2022 The FeatHub Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.feathub.flink.udf;
+
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.operators.OneInputStreamOperator;
+import org.apache.flink.types.Row;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/** Utility methods for Java UDFs. */
+public class JavaUdfUtils {
+    /**
+     * Evaluates a Java UDF against a DataStream.
+     *
+     * @param stream DataStream containing input data for the UDF.
+     * @param className The classpath of the UDF to be evaluated. Must be a subclass of {@link
+     *     OneInputStreamOperator}.
+     * @param parameters The constructor parameters of the UDF.
+     * @return DataStream containing output data of the UDF.
+     */
+    @SuppressWarnings({"unchecked"})
+    public static DataStream<Row> evalJavaUdf(
+            DataStream<Row> stream, String className, List<?> parameters) throws Exception {
+        // TODO: Support cases in which the input and output type differs.
+        return stream.transform(
+                        "JavaUDF",
+                        stream.getType(),
+                        (OneInputStreamOperator<Row, Row>) instantiate(className, parameters))
+                // TODO: Generify the configuration of parallelism.
```

Comment:
Rate and Count metrics are both computed using `SlidingWindowTransform` with `group_by_keys = ()`, which the corresponding DataStream should already have parallelism = 1.

Instead of giving developers the idea that we are changing the parallelism to 1 here, it might be more readable to verify that the input `stream` already has parallelism=1.

In the future, when we have metrics whose DataStream can have parallelism > 1, we might find it reasonable to set the parallelism to be same as `stream` here.

Reply:
When SlidingWindowTransform has group_by_keys = (), the current implementation would still generate a result datastream whose parallelism is larger than 1. This is also reasonable given that records with the same key can be pre-aggregated on different subtasks at first for better performance, according to the JavaDoc of `org.apache.flink.api.common.functions.AggregateFunction`.

Reply:
Would it be more readable to verify that the input stream already has parallelism=1?